frontend/src/lib/WidgetStateManager.test.ts

```diff
@@ -211,4 +211,122 @@ describe("Widget State Manager", () => {
       expect(widgetMgr.getIntArrayValue(MOCK_DATA)).toStrictEqual(values)
     })
   })
+
+  describe("submitForm", () => {
+    it("doesn't submit the form if the form id is an empty string", () => {
+      const invalidFormId = ""
+
+      expect(widgetMgr.submitForm(invalidFormId)).toBeUndefined()
+    })
+
+    it("doesn't submit the form if the form doesn't have pending widget states", () => {
+      const formId = "TEST_ID"
+
+      expect(widgetMgr.submitForm(formId)).toBeUndefined()
+    })
```

Comment:
For these tests, we should also be explicitly checking that WidgetStateManager is not calling `sendBackMsg`:

```javascript
const invalidFormId = ""
expect(widgetMgr.submitForm(invalidFormId)).toBeUndefined()
expect(sendBackMsg).not.toHaveBeenCalled()
```

(Or use the `sendUpdateWidgetsMessageSpy` technique you use below)

Reply:
Done